src/Redux/fireRequest.tsx

```diff
@@ -152,6 +152,8 @@ export const fireRequest = (
           if (error.response.status > 400 && error.response.status < 500) {
             if (error.response.data && error.response.data.detail) {
               if (error.response.data.code === "token_not_valid") {
+                //store the path before session expiry in local storage
+                localStorage.setItem("lastPath", window.location.href);
```

Comment:
This behavior shouldn't be coupled with `fireRequest`. Instead just replace all navigates to `/session-expired` to `/session-expired?redirect=<url encoded location href>`.

fireRequest is deprecated.

```suggestion
```